src/main/scala/viper/server/VerificationWorker.scala

```diff
@@ -72,13 +60,40 @@ class VerificationWorker(private val reporterActor: ActorRef,
   }
 
   private var backend: ViperBackend = _
+  implicit val executionContext = ExecutionContext.global
+
+  // Implementation of the Reporter interface used by the backend.
+  class ActorReporter(private val actor_ref: ActorRef,
+                      val tag: String) extends viper.silver.reporter.Reporter {
+
+    val name = s"ViperServer_$tag"
+    var current_offer: Future[QueueOfferResult] = null
+
+    /** Sends massage to the attached actor.
+      *
+      * The actor receving this message offer it to a queue. This offering returns a Future, which will eventually
+      * indicate whether or not the offer was successful. This method is blocking, as it waits for the successful
+      * completion of such an offer.
+      * */
+    def report(msg: reporter.Message): Unit = {
+      implicit val askTimeout: Timeout = Timeout(5000 milliseconds)
```

Comment:
Could we add these constants to the server configuration?

Reply:
This particular timeout code exists i several places where actors ask something (in fact, it is required). I took over the threshold form existing code. Should every such instance be parametrised?

Reply:
I think it's generally a good idea to define all constants in the same place, rather than to hardcode them, especially if they could be reused in multiple places in the code. This way it's easier to document what parameters the server depends on (all configurations have descriptors that explain what they mean). 

I know that this was not always the case in the original code. 

This is a low priority task.